### DISCUSSION TITLE
SFTP reader Errors in Dashboard but not in channel

### DISCUSSION THREAD

**Discussion by Snowskier** [nextgenhealthcare/connect#5333] [Q&A]
Mirth version 4.0.1
We have an SFTP reader that picks up PDF files and sends them to a downstream channel to deliver to another system via SFTP. We are using attachments since these PDF files can be bloated (Patient charts). We are seeing the following in the dashboard however no errors show up in either the source channel or the destination channel.

[2022-08-03 03:39:42,604] ERROR (com.mirth.connect.connectors.file.FileReceiver:27 ): Error processing file in channel: 67d9cca7-6561-446c-8e7a-d934d180aaf3
com.jcraft.jsch.JSchException: Session.connect: java.net.SocketTimeoutException: Read timed out

[2022-08-03 03:39:52,612] ERROR (com.mirth.connect.connectors.file.FileReceiver:27 ): Unable to dispatch message to channel 67d9cca7-6561-446c-8e7a-d934d180aaf3. File: /prod/pdf/DATE-08_02_22_CSN-464112133_EPE URGENCY ROOM.PDF
com.jcraft.jsch.JSchException: Session.connect: java.net.SocketTimeoutException: Read timed out

Are these caused by the Mirth downstream channel being busy as the reader attempts to dispatch the message to it? We verified that all files were accounted for.

Thanks in advance

**Comment by rjkroll** [comment]
That a difficult one to figure out what is actually causing it.......but, it might actually be related to this bug in the JSCH library.    What version of Mirth are you running?

https://github.com/nextgenhealthcare/connect/issues/4080

**Reply by Snowskier** [reply]
Thanks for the reply. Version 4.0.1 we are upgrading to 4.1.0 next monday

**Comment by jonbartels** [comment]
` java.net.SocketTimeoutException: Read timed out` makes me think its an ordinary network timeout. 

> Are these caused by the Mirth downstream channel being busy as the reader attempts to dispatch the message to it? We verified that all files were accounted for.

How are you dispatching the message and does the downstream channel have a lot of logic in the source transformers?

**Reply by Snowskier** [reply]
Thanks for the reply 
We are using a channel writer to dispatch to the downstream channel. No logic at all simply a passthru
